### PR TITLE
Splitting litigator claim pages

### PR DESCRIPTION
**Pull request by lostie**
💎  [Page splitting for Litigator final fee](https://www.pivotaltracker.com/story/show/154799214)
💎  [Page splitting for Litigator Interim Fee](https://www.pivotaltracker.com/story/show/154799221)
💎  [Page splitting for Litigator Transfer Fee](https://www.pivotaltracker.com/story/show/154799228)

Also:

* Change from *numbered steps* to **named steps**. Numbered steps at a given time can represent any sort of step, which makes it quite hard to understand what step was taken by a user in the past.

### CODE REVIEW

File: app/models/claim/interim_claim.rb [hunk 1]
```diff
@@ -76,6 +76,10 @@ class InterimClaim < BaseClaim
     accepts_nested_attributes_for :interim_fee, reject_if: :all_blank, allow_destroy: false
     accepts_nested_attributes_for :warrant_fee, reject_if: :all_blank, allow_destroy: false
 
+    def submission_stages
+      %i[case_details defendants offence_details fees]
```
Comment (colinbruce) [inline app/models/claim/interim_claim.rb:80]: Nice 👍

File: app/validators/claim/advocate_claim_sub_model_validator.rb [hunk 2]
```diff
@@ -1,24 +1,24 @@
 class Claim::AdvocateClaimSubModelValidator < Claim::BaseClaimSubModelValidator
   def has_one_association_names_for_steps
-    [
-      [],
-      [],
-      [],
-      %i[
+    {
+      case_details: [],
```
Comment (colinbruce) [inline app/validators/claim/advocate_claim_sub_model_validator.rb:4]: Named steps make this much easier to read too

**Review by colinbruce — APPROVED** [review]
LGTM 👍

File: app/views/external_users/litigators/claims/_form.html.haml [hunk 3]
```diff
@@ -4,41 +4,4 @@
...
-      = render partial: 'external_users/claims/warrant_fee/fields', locals: { f: f }
-
-      = render partial: 'external_users/claims/supporting_documents/fields', locals: { f: f }
-
-      = render partial: 'external_users/claims/supporting_evidence_checklist/fields', locals: { f: f }
-
-      = render partial: 'external_users/claims/additional_information/fields', locals: { f: f }
-
-      = render partial: 'external_users/claims/buttons/continue', locals: { f: f, commit: 'commit_submit_claim' }
+    = render partial: "#{claim.current_step}_form_step", locals: { claim: claim, f: f }
```
Comment (jsugarman) [inline app/views/external_users/litigators/claims/_form.html.haml:7]: cool

**Review by jsugarman — APPROVED** [review]
looks good 👍

**Pull request merged**